modules/indexer/code/git.go

```diff
@@ -161,15 +162,25 @@ func nonGenesisChanges(ctx context.Context, repo *repo_model.Repository, revisio
 		default:
 			log.Warn("Unrecognized status: %c (line=%s)", status, line)
 		}
-	}
 
-	cmd := git.NewCommand(ctx, "ls-tree", "--full-tree", "-l").AddDynamicArguments(revision).
-		AddDashesAndList(updatedFilenames...)
-	lsTreeStdout, _, err := cmd.RunStdBytes(&git.RunOpts{Dir: repo.RepoPath()})
-	if err != nil {
-		return nil, err
+		// According to https://learn.microsoft.com/en-us/troubleshoot/windows-client/shell-experience/command-line-string-limitation#more-information
+		// the command line length should less than 8191 characters, assume filepath is 256, then 8191/256 = 31, so we use 30
+		if (i%30 == 0 || i == len(lines)-1) && len(updatedFilenames) > 0 {
```

Comment:
It is wrong here.

If many files are ignored, then `i%30==0` is true, but `updatedFilenames` is empty

Reply:
e16bb63be589f15bf32262080ae3fe1b0c574eaf